x-pack/plugin/ml-feature-index-builder/src/test/java/org/elasticsearch/xpack/ml/featureindexbuilder/job/SourceConfigTests.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.featureindexbuilder.job;
+
+import org.elasticsearch.common.io.stream.Writeable.Reader;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.search.aggregations.bucket.composite.CompositeValuesSourceBuilder;
+import org.elasticsearch.search.aggregations.bucket.composite.TermsValuesSourceBuilder;
+import org.elasticsearch.search.sort.SortOrder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SourceConfigTests extends AbstractSerializingFeatureIndexBuilderTestCase<SourceConfig> {
+
+    public static SourceConfig randomSourceConfig() {
+        int numSources = randomIntBetween(1, 10);
```

Comment:
Is it valid for the number of sources to be 0? If not maybe the `SourceConfig` class itself should validate that it has at least 1?

Reply:
the for-loop below counts from `0`, so there will be at least 1 source for this test.

`SourceConfig` lacks validation at the moment, to be addressed in an upcoming PR.